src/Microsoft.Identity.Client/TokenCache.MigrationAid.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Identity.Client.Cache;
+
+namespace Microsoft.Identity.Client
+{
+    public sealed partial class TokenCache : ITokenCacheInternal
+    {
+        /// <summary>
+        /// Gets or sets the flag indicating whether the state of the cache has changed.
+        /// MSAL methods set this flag after any change.
+        /// Caller applications should reset the flag after serializing and persisting the state of the cache.
+        /// </summary>
+        [Obsolete("Please use the equivalent flag TokenCacheNotificationArgs.HasStateChanged, " +
+        "which indicates if the operation triggering the notification is modifying the cache or not." +
+        " Setting the flag is not required.")]
+        public bool HasStateChanged
+        {
+            get => _hasStateChanged;
+            set => _hasStateChanged = value;
+        }
+
+#if !ANDROID_BUILDTIME && !iOS_BUILDTIME
+
+        /// <summary>
+        /// Serializes the entire token cache in both the ADAL V3 and unified cache formats.
+        /// </summary>
+        /// <returns>Serialized token cache <see cref="CacheData"/></returns>
+        /// <remarks>
+        /// <see cref="SerializeMsalV3"/>/<see cref="DeserializeMsalV3"/> is compatible with other MSAL libraries such as MSAL for Python and MSAL for Java.
+        /// </remarks>
+        [Obsolete("This is expected to be removed in MSAL.NET v3 and ADAL.NET v5. We recommend using SerializeMsalV3/DeserializeMsalV3. Read more: https://aka.ms/msal-net-3x-cache-breaking-change", false)]
```

Comment:
> MSAL.NET v3 and ADAL.NET v5 [](start = 52, length = 28)

perhaps update version numbers, at that will not happen... perhaps in the next versions though

Reply:
fixed